src/daemon/collector/content/collect.js

```diff
@@ -490,51 +487,46 @@ function findFrontAdsWithHiddenLettersSiblings(){
 
 function getFrontAdFrames() {
   let a_links = document.getElementsByTagName('a');
-  let {links, newStyle} = filterFrontAds(a_links);
+  let {links, layoutStyle} = filterFrontAds(a_links);
 
-  links = links.concat(getFrontAdsByClass())
+  links = links.concat(getFrontAdsByClass(layoutStyle))
   links = links.concat(findFrontAdsWithHiddenLetters())
   links = links.concat(findFrontAdsWithHiddenLettersSiblings())
   links = [...new Set(links)]
 
-  let frontAds = [];
-  for (let i = 0; i < links.length; i++) {
-    let frame;
-    if (newStyle) {
-      frame = getParentAdDivNewStyle(links[i]);
-    } else {
-      frame = getParentAdDiv(links[i]);
-    }
-    frontAds.push(frame);
-  }
-  return {frontAds: filterCollectedAds(frontAds, newStyle), newStyle};
+  const frontAds = links.map(link => layoutStyle === "FB5" ? getParentAdDivNewStyle(link) : getParentAdDiv(link));
+  return {frontAds: filterCollectedAds(frontAds, layoutStyle), layoutStyle};
+}
+
+function generateRelatedField(id) {
+  return id + "_" + uuidv4();
 }
 
 // works - collecting and logging ads (old, new)
 // diff - SF, main app
-function processFrontAd(frontAd, newStyle) {
+function processFrontAd(frontAd, layoutStyle) {
   // console.log('processFrontAd', frontAd, COLLECTED)
   frontAd.className += " " + "ad_collected";
   const raw_ad = $(frontAd).html();
   let parent_id;
-  if (newStyle) {
+  if (layoutStyle === "FB5") {
     parent_id = $(frontAd).attr('aria-labelledby');
     COLLECTED.push(parent_id);
   } else {
     parent_id = $(frontAd).attr('id');
   }
-  // console.log('!!!!!raw_ad ------ collected', newStyle, frontAd)
-  // console.log('!!!!!raw_ad ------ parent_id', newStyle, parent_id)
+  // console.log('!!!!!raw_ad ------ collected', layoutStyle, frontAd)
+  // console.log('!!!!!raw_ad ------ parent_id', layoutStyle, parent_id)
 
   // ----- Temporarily collect ads even if rationales are not there ----- //
   const container = $(raw_ad);
   let fbStoryId;
-  if (newStyle) {
-    fbStoryId = parent_id;
+  if (layoutStyle === "FB5") {
+    fbStoryId = generateRelatedField(parent_id);
```

Comment:
what is this id going to look like?

Reply:
somewhat like this "jsc_c_4_317c3fd7-3e83-47f6-87e1-e40f8f70eb50"

Reply:
Does the `jsc_c_4_` bit still have purpose?